### PR TITLE
Lint all the things

### PR DESCRIPTION
**Pull request by live627**
Someone should enable builds on Travis CI

### CODE REVIEW

File: Sources/Subs.php [hunk 1]
```diff
@@ -5806,7 +5806,10 @@ function build_regex($strings, $delim = null)
 				$sub_regex = $index_to_regex($value, $delim);
 
 				if (count(array_keys($value)) == 1)
-					$new_key .= explode('(?'.'>', $sub_regex)[0];
```
Comment (colinschoen) [inline Sources/Subs.php:5809]: What is this change for?

**Review by colinschoen — CHANGES_REQUESTED** [review]
Not sure what Subs is being changed for.

File: Sources/Subs.php (hunk 1 above)
Comment (live627) [reply Sources/Subs.php:5809]: PHP 5.3 compatibility.

**Comment by colinschoen** [comment]
@CoreISP or @Oldiesmann (or whoever has owners access) can we enable the travis CI extension.

**Review by colinschoen — APPROVED** [review]

**Comment by colinschoen** [comment]
Bump

**Comment by jdarwood007** [comment]
@colinschoen,

I am asking @Oldiesmann to setup travis.  Although at this point I'm not sure what use it would be without some proper tools to test with.  Do you have stuff in mind?  We do have a unreleased PHP unit that was for testing BBC and a few other things sitting around.  I think if you are willing to take on that PHP unit, we could get it updated, released and pushed for Travis CI to use.

**Comment by colinschoen** [comment]
Thanks, yes willing to take that on.

**Comment by jdarwood007** [comment]
https://travis-ci.org/SimpleMachines/SMF2.1

@Oldiesmann has set it up btw.

**Pull request merged**